### PR TITLE
Add permanent kill switch and redirect stderr and stdout for up check

### LINKED ISSUE #5: Maybe use iptables to implement kill switch option?
Since somehow the kill switch was not working for me inside the docker container, I looked at the `protonvpn-cli` source code and saw they are using plain iptables as well. So I modified the entrypoint to block everything except to LAN and the `proton0` device. 

Keep in mind I have hardcoded my container subnet to `172.26.0.0/16` in my `docker-compose.yml` for this to work, but it should be very doable to fix this. I'm just providing this is a proof of concept, please let me know if I missed anything in the rules, from my understanding, this should block all traffic outside LAN on the main adapter and only allow that traffic through `proton0`.

```
iptables -P INPUT DROP
iptables -P FORWARD DROP
iptables -P OUTPUT DROP
iptables -A INPUT -i lo -j ACCEPT
iptables -A INPUT -i proton0 -j ACCEPT
iptables -A INPUT -i proton0 -m state --state RELATED,ESTABLISHED -j ACCEPT
iptables -A INPUT -p udp -m udp --sport 1194 -j ACCEPT
iptables -A INPUT -s 172.26.0.0/16 -i eth0 -j ACCEPT
iptables -A INPUT -s 10.8.0.0/16 -i eth0 -j ACCEPT
iptables -A OUTPUT -o lo -j ACCEPT
iptables -A OUTPUT -o proton0 -j ACCEPT
iptables -A OUTPUT -o proton0 -m state --state RELATED,ESTABLISHED -j ACCEPT
iptables -A OUTPUT -p udp -m udp --dport 1194 -j ACCEPT
iptables -A OUTPUT -d 172.26.0.0/16 -o eth0 -j ACCEPT
iptables -A OUTPUT -d 10.8.0.0/16 -o eth0 -j ACCEPT
```

Since VPN users often do want their traffic to only go through the VPN tunnel, I imagine an environment variable might that triggers this might be useful maybe? I currently just have the above rules in the entrypoint script.

### PR DESCRIPTION
**Pull request by ProgramComputer**
[linux-cli-community](https://github.com/Rafficer/linux-cli-community) may be unmantained. Resolves #5 by adding package `iptables`.

Adds experimental kill-switch separate from the cli to container. Permanent kill switch is set to default. The killswitch is ip-agnostic relying on processes and interfaces(eth0 and proton0) and supports ipv6 from using nftables.

 Below was the initial nfttable configure file but took space using sets even when size of 65535 is sufficient above [128](http://www.privoxy.org/3.0.21/user-manual/config.html), it would hinder configurability. Privoxy also [recommends](https://www.privoxy.org/user-manual/installation.html) not to run as root. Finally manually enabling the cli killswitch may cause unexpected results.
Not tested in swarm mode. 
```
#!/bin/env net -f
table inet filter {
  set eth0_clients4 {                                                 
        type ipv4_addr  
  size 65535                                              
        flags timeout         
gc-interval 1d                           
    }           

set eth0_clients6 {                                                 
        type ipv6_addr    
  size 65535                                            
        flags timeout          
gc-interval 1d                          
    }           

    chain input {
        type filter hook input priority 0; policy drop;
	iif lo accept
        # Add IP addresses to the eth0_clients set when clients connect
        add @eth0_clients4 {ip saddr} 
        add @eth0_clients6 {ip6 saddr} 

        # Allow incoming traffic from eth0 to Privoxy (IPv4)
        iifname "eth0" ip saddr @eth0_clients4 accept

        # Allow incoming traffic from eth0 to Privoxy (IPv6)
        iifname "eth0" ip6 saddr @eth0_clients6 accept

        # Allow established and related traffic
        ct state established,related accept

        # Drop everything else
        drop
    }

   chain output {
    type filter hook output priority 0; policy drop;
    
	oif lo accept
	oif eth0 skuid root accept

          # Allow established and related traffic
	ct state established,related accept
        
# Drop eth0 loopbacks
	ip saddr @eth0_clients4 drop
	ip6 saddr @eth0_clients6 drop
    # Remove IP addresses from the eth0_clients set when traffic goes out on proton0 (IPv4)
     oifname proton0 ip daddr @eth0_clients4 
	delete @eth0_clients4  {ip daddr }

    # Remove IP addresses from the eth0_clients set when traffic goes out on proton0 (IPv6)
     oifname proton0 ip6 daddr @eth0_clients6 
	delete @eth0_clients6  {ip6 daddr }

    # Allow outgoing traffic on the proton0 interface 
    oifname "proton0" accept

    # Drop everything else
    drop
}

}

```

Minor change was from seeing #36 and preventing some redundant output.

### CODE REVIEW

File: pvpn-cli/killswitch.conf [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/env net -f
```
Comment (walterl) [inline pvpn-cli/killswitch.conf]: I haven't used `nft` like this before, but this line should look like below, right?

```suggestion
#!/bin/env nft -f
```

File: pvpn-cli/killswitch.conf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+table inet filter {
+
+    chain input {
+        type filter hook input priority 0; policy drop;
+        # Allow loopback
+        iif lo accept
+
+        # Allow new, established, and related traffic
+        ct state new,established,related accept
```
Comment (walterl) [inline pvpn-cli/killswitch.conf:11]: Why do we want to allow incoming connections?

File: README.md [hunk 3]
```diff
@@ -78,17 +78,8 @@ be routed over the same VPN connection.
...
-
-**UPDATE:** This is no longer an issue, because Docker now allows
-`/etc/resolv.conf` to be updated while a container is running. It's recreated
-by Docker on container restart, but that doesn't matter, since ProtonVPN (and
-`DNS_SERVERS_OVERRIDE`) will modify it during startup.
-
-~Docker prevents containers from changing the servers used for DNS lookups, after startup. This prevents ProtonVPN from using its own leak protecting DNS server. In fact, at the moment it causes a non-fatal error in `protonvpn`.~
-
-~Ensure that you're using privacy respecting DNS servers on your Docker host, or manually specify secure servers for the container via [`--dns` options](https://docs.docker.com/config/containers/container-networking/#dns-services).~
-
```
Comment (walterl) [inline README.md]: I don't think this section should be removed, because [ProtonVPN's DNS leak protection](https://protonvpn.com/support/dns-leaks-privacy/) is a separate feature from the [killswitch](https://protonvpn.com/support/what-is-kill-switch/).

Even if it was the same thing, users could reasonably want to _not_ use ProtonVPN's DNS servers, even when using ProtonVPN, so I'd rather keep this information in.

File: app/proton-privoxy/run [hunk 4]
```diff
@@ -63,6 +63,11 @@ if [ -n "$HOST_NETWORK" ]; then
...
 
+# Set Killswitch
+if [ $PVPN_KILL -eq 1 ]; then
+nft -f $PVDIR/killswitch.conf
+fi
+
 # Start Privoxy
 ln -s /etc/privoxy/templates /app/proton-privoxy/
-exec privoxy --no-daemon
+exec su privoxyu -c 'privoxy --no-daemon'
```
Comment (walterl) [inline app/proton-privoxy/run:73]: This is great, thanks! :heart:

File: app/proton-privoxy/run [hunk 5]
```diff
@@ -63,6 +63,11 @@ if [ -n "$HOST_NETWORK" ]; then
 	ip route add "$HOST_NETWORK" via "$gw"
 fi
 
+# Set Killswitch
+if [ $PVPN_KILL -eq 1 ]; then
+nft -f $PVDIR/killswitch.conf
```
Comment (walterl) [inline app/proton-privoxy/run]:
```suggestion
	nft -f $PVDIR/killswitch.conf
```

**Review by walterl — CHANGES_REQUESTED** [review]
Thanks so much for this valuable contribution. :heart:

Left some comments for minor changes, and a question or two that I'm unsure about.

Please also bump the version in the Dockerfile to 0.5.0 in all these places:

```
❯ ag 0.4
test/docker_secrets_test/test.sh:3:#docker build -t walt3rl/proton-privoxy:0.4.2-dev ../..
Dockerfile:3:LABEL version="0.4.2"
test/docker_secrets_test/docker-compose.yml:4:    image: walt3rl/proton-privoxy:0.4.2-dev
.github/workflows/docker.yml:39:            walt3rl/proton-privoxy:0.4.2
```

File: pvpn-cli/killswitch.conf (hunk 2 above)
Comment (ProgramComputer) [reply pvpn-cli/killswitch.conf:11]: The first line can be removed; I left it for future testing purposes.
The second line is to initially connect with privoxy; otherwise no incoming [connection](https://wiki.nftables.org/wiki-nftables/index.php/Matching_connection_tracking_stateful_metainformation#ct_state_-_conntrack_state) can be made except outgoing from within the proxy. Try replacing with ` iif eth0 skuid privoxyu accept`, but I think the functionality won't change significantly.

Comment (walterl) [reply pvpn-cli/killswitch.conf:11]: I understand what this does, but not clear on why we would and to allow _incoming_ connections at all. The connection to setup the VPN is outgoing, as is all connections using the VPN, right?

In fact, shouldn't the `new` on this line be added to the corresponding `output` rule? Otherwise no new connection can be made _from_ this container.

Comment (ProgramComputer) [reply pvpn-cli/killswitch.conf:11]: Yes, outgoing connections can already be made as proton0 allows it, but to connect to privoxy as a proxy incoming connection of some form has to be allowed; otherwise, it's not a proxy, but a direct VPN connection from within. I recommend testing your edge cases and letting me know.

Comment (walterl) [reply pvpn-cli/killswitch.conf:11]: Right. That makes sense, but it still seems odd that _all_ incoming connections are allowed, and no outgoing ones. I'll test it out a bit.
